numpy/__init__.py

```diff
@@ -107,6 +107,8 @@
 import warnings
 
 from ._globals import _NoValue, _CopyMode
+from . import exceptions
+# Note that the following names are imported explicitly for backcompat:
```

Comment:
Do we really have to export Python errors/warnings in the NumPy top-level namespace? If so those should be deprecated.

Reply:
I agree we should follow-up with deprecating or at least hiding them from the main namespace.   We could move the "namespace", I held off with that in the first PR, but the only thing that might notice is unpickling in an older version, and that is probably not worth the trouble for exceptions.